numpy/core/src/multiarray/arrayobject.c

```diff
@@ -112,7 +112,7 @@ PyArray_SetUpdateIfCopyBase(PyArrayObject *arr, PyArrayObject *base)
      * references.
      */
     ((PyArrayObject_fields *)arr)->base = (PyObject *)base;
-    PyArray_ENABLEFLAGS(arr, NPY_ARRAY_UPDATEIFCOPY);
+    PyArray_ENABLEFLAGS(arr, NPY_ARRAY_UPDATEIFCOPY_CLEAR_B4_EXIT);
```

Comment:
Ugh, this is dubious. `PyArray_SetUpdateIfCopyBase` is public, so there might be third-party code that uses it and expects the GC to resolve the `UPDATEIFCOPY`. I guess we need to make `PyArray_SetUpdateIfCopyBase` keep using `NPY_ARRAY_UPDATEIFCOPY` and issue a deprecation warning, and add a new function `PyArray_SetUpdateIfCopyBaseWithExplicitResolve` or something like that.